libcontainer/container_linux.go

```diff
@@ -186,8 +186,17 @@ func (c *linuxContainer) Set(config configs.Config) error {
 	if status == Stopped {
 		return newGenericError(fmt.Errorf("container not running"), ContainerNotRunning)
 	}
+	if err := c.cgroupManager.Set(&config); err != nil {
+		// Set configs back
+		if err2 := c.cgroupManager.Set(c.config); err2 != nil {
```

Comment:
I am not sure about external users of libcontainer, but runc's own `update.go` modifies container config in place, meaning `&config` and `c.config` is exactly the same thing, meaning this "rollback" does not make sense.

@hqhq do you remember why you introduced it?

Reply:
@kolyshkin Libcontainer was supposed to be used without runc, so the config parameter can be different from c.config, if I remember right, before I added `runc update`, docker was using this `Set` function of libcontainer directly for `docker update`, this rollback was valid at that time.